File: client/admin/src/main/java/com/reservation/admin/terms/service/dto/SearchTermsResult.java
package com.reservation.admin.terms.service.dto;

import java.time.LocalDateTime;

import com.querydsl.core.annotations.QueryProjection;
import com.reservation.domain.terms.enums.TermsCode;
import com.reservation.domain.terms.enums.TermsStatus;
import com.reservation.domain.terms.enums.TermsType;

import lombok.Getter;

@Getter
public class SearchTermsResult {
	Long id;
	TermsCode code;
	String title;
	TermsType type;
	TermsStatus status;
	Integer version;
	Boolean isLatest;
	LocalDateTime exposedFrom;
	LocalDateTime exposedToOrNull;
	Integer displayOrder;
	LocalDateTime createdAt;

	@QueryProjection
	public SearchTermsResult(
		Long id,
		TermsCode code,
		String title,
		TermsType type,
		TermsStatus status,
		Integer version,
		Boolean isLatest,
		LocalDateTime exposedFrom,
		LocalDateTime exposedToOrNull,
		Integer displayOrder,
		LocalDateTime createdAt
	) {
		this.id = id;
		this.code = code;
		this.title = title;
		this.type = type;
		this.status = status;
		this.version = version;
		this.isLatest = isLatest;
		this.exposedFrom = exposedFrom;
		this.exposedToOrNull = exposedToOrNull;
		this.displayOrder = displayOrder;
		this.createdAt = createdAt;
	}
}

File: client/admin/src/main/java/com/reservation/admin/terms/repository/dto/SearchTermsResult.kt
package com.reservation.admin.terms.repository.dto

import com.querydsl.core.annotations.QueryProjection
import com.reservation.domain.terms.enums.TermsCode
import com.reservation.domain.terms.enums.TermsStatus
import com.reservation.domain.terms.enums.TermsType
import lombok.Getter
import java.time.LocalDateTime

@Getter
class SearchTermsResult @QueryProjection constructor(
    var id: Long,
    var code: TermsCode,
    var title: String,
    var type: TermsType,
    var status: TermsStatus,
    var version: Int,
    var isLatest: Boolean,
    var exposedFrom: LocalDateTime,
    var exposedToOrNull: LocalDateTime,
    var displayOrder: Int,
    var createdAt: LocalDateTime
)
